### ISSUE TITLE
cleanup conditional compilation for responses

### ISSUE THREAD

**Issue by Parth** [lockbook/lockbook#1944]

**Comment by tvanderstad** [comment]
egui client isn't building on macos
```
~/code/lockbook/lockbook/clients/egui smail/apple-paste-images ❯ cargo check
warning: profiles for the non root package will be ignored, specify profiles at the workspace root:
package:   /Users/travis/Code/lockbook/lockbook/libs/core/Cargo.toml
workspace: /Users/travis/Code/lockbook/lockbook/Cargo.toml
    Checking egui_editor v0.8.0 (/Users/travis/Code/lockbook/lockbook/libs/editor/egui_editor)
    Checking lockbook-egui v0.8.0 (/Users/travis/Code/lockbook/lockbook/clients/egui)
error[E0308]: mismatched types
  --> clients/egui/src/account/tabs/markdown.rs:41:24
   |
41 |                 if let Some(title) = &res.potential_title {
   |                        ^^^^^^^^^^^   -------------------- this expression has type `&*const i8`
   |                        |
   |                        expected `*const i8`, found `Option<_>`
   |
   = note: expected raw pointer `*const i8`
                     found enum `std::option::Option<_>`

For more information about this error, try `rustc --explain E0308`.
error: could not compile `lockbook-egui` (bin "lockbook-egui") due to previous error
```

**Comment by Parth** [comment]
fixed in #2280
